core/chains/evm/client/send_only_node_test.go

```diff
@@ -0,0 +1,210 @@
+
+	node := evmclient.NewSendOnlyNode(lggr, *url, name, chainID)
+	assert.NotNil(t, node)
+
+	// Must contain name & url with redacted password
+	assert.Contains(t, node.String(), fmt.Sprintf("%s:%s", name, redacted))
+	assert.Equal(t, node.ChainID(), chainID)
+}
+
+func TestStartSendOnlyNode(t *testing.T) {
+	t.Run("Start with Random ChainID", func(t *testing.T) {
+		chainID := testutils.NewRandomEVMChainID()
+		r := chainIDResp{chainID.Int64(), nil}
+		url := r.newHTTPServer(t)
+		lggr, observedLogs := logger.TestLoggerObserved(t, zap.WarnLevel)
+		s := evmclient.NewSendOnlyNode(lggr, *url, t.Name(), chainID)
+		//n := iNode.(*evmclient.sendOnlyNode)
+		defer s.Close()
+		err := s.Start(testutils.Context(t))
+		assert.NoError(t, err)                 // No errors expected
+		assert.Equal(t, 0, observedLogs.Len()) // No warnings expected
+	})
+
+	t.Run("Start with ChainID=0", func(t *testing.T) {
+		lggr, observedLogs := logger.TestLoggerObserved(t, zap.WarnLevel)
+		chainID := testutils.FixtureChainID
+		r := chainIDResp{chainID.Int64(), nil}
+		url := r.newHTTPServer(t)
+		s := evmclient.NewSendOnlyNode(lggr, *url, t.Name(), testutils.FixtureChainID)
+
+		defer s.Close()
+		err := s.Start(testutils.Context(t))
+		assert.NoError(t, err)
+		// getChainID() should return Error if ChainID = 0
+		// This should get converted into a warning from Start()
+		testutils.WaitForLogMessage(t, observedLogs, "ChainID verification skipped")
+	})
+}
+
+func createSignedTx(t *testing.T, chainID *big.Int, nonce uint64, data []byte) *types.Transaction {
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+	sender, err := bind.NewKeyedTransactorWithChainID(key, chainID)
+	require.NoError(t, err)
+	tx := types.NewTransaction(
+		nonce, sender.From,
+		assets.Ether(100),
+		21000, big.NewInt(1000000000), data,
+	)
+	signedTx, err := sender.Signer(sender.From, tx)
+	require.NoError(t, err)
+	return signedTx
+}
+
+type MockTxSender struct {
+	mock.Mock
+}
+
+// ChainID provides a mock function for ethclient.SendTransaction()
+func (_m MockTxSender) ChainID(ctx context.Context) (*big.Int, error) {
+	ret := _m.Called()
+
+	var r0 *big.Int
+	if rf, ok := ret.Get(0).(func(context.Context) *big.Int); ok {
+		r0 = rf(ctx)
+	} else {
+		if ret.Get(0) != nil {
+			r0 = ret.Get(0).(*big.Int)
+		}
+	}
+
+	var r1 error
+	if rf, ok := ret.Get(2).(func(context.Context) error); ok {
+		r1 = rf(ctx)
+	} else {
+		r1 = ret.Error(2)
+	}
+	return r0, r1
+}
+
+// SendTransaction provides a mock function for ethclient.SendTransaction()
+func (_m MockTxSender) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+	ret := _m.Called(ctx, tx)
+
+	var r0 error
+	if rf, ok := ret.Get(0).(func(context.Context, *types.Transaction) error); ok {
+		r0 = rf(ctx, tx)
+	} else {
+		r0 = ret.Error(0)
+	}
+
+	return r0
+}
+
+type MockBatchSender struct {
+	mock.Mock
+}
+
+func (_m MockBatchSender) BatchCallContext(ctx context.Context, b []rpc.BatchElem) error {
```

Comment:
Should not this be auto-generated and kept in a separate file, similar to all other mocks we have?